### PR TITLE
optimizing docker build process and using smaller base image

### PR DESCRIPTION
**Pull request by denniswebb**
This PR includes changing the kexpand docker image to be based on alpine:3.4 over debian:jessie.  This resulted in a size decrease from 130MB down to just 8MB.

The build process is now completed using golang:1.7 image instead of debian:jessie and installing go onto that container.  This takes a long step out of the process and reduces initial runs of `make images` by over half in my tests.

current method
```
Successfully built 16c67599fd0c
      280.54 real         0.16 user         0.22 sys
~/work/src/github.com/kopeio/kexpand (master) $ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
kopeio/kexpand      latest              16c67599fd0c        2 minutes ago       129.2 MB
kexpand-builder     latest              ca79a5c903b4        2 minutes ago       667.2 MB
debian              jessie              7b0a06c805e8        2 weeks ago         123 MB
```

proposed method:
```
Successfully built f90b43e855ee
      129.63 real         0.18 user         0.21 sys
~/work/src/github.com/kopeio/kexpand (feature/docker-optimize) $ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
kopeio/kexpand      latest              f90b43e855ee        7 minutes ago       7.923 MB
golang              1.7                 f69c27b2f59a        13 days ago         672.4 MB
alpine              3.4                 baa5d63471ea        2 weeks ago         4.803 MB
```

### CODE REVIEW

File: images/kexpand/onbuild.sh [hunk 1]
```diff
@@ -7,7 +7,10 @@ mkdir -p /go/src/github.com/kopeio
 ln -s /src/ /go/src/github.com/kopeio/kexpand
 
 cd /go/src/github.com/kopeio/kexpand
-make gocode
+if [ ! -d vendor ]; then
```
Comment (denniswebb) [inline images/kexpand/onbuild.sh:10]: added so a new user without experience using govendor can download the source and run `make image` without requiring a local installation of go just to retrieve libraries.

File: Makefile [hunk 2]
```diff
@@ -8,16 +8,15 @@ endif
 gocode:
 	GO15VENDOREXPERIMENT=1 go install -ldflags "-X main.BuildVersion=${VERSION}" github.com/kopeio/kexpand
 
+gocode_docker:
```
Comment (denniswebb) [inline Makefile:11]: Added to allow a completely statically linked linux binary

**Comment by justinsb** [comment]
I'm also playing with bazel, which has a really cool feature that it can build Docker images directly, but has the flipside that you have to install bazel.

**Comment by justinsb** [comment]
(see #13)

**Review by justinsb — APPROVED** [review]
LGTM - thanks!

**Pull request merged**